### PR TITLE
Fix OpenMP again

### PR DESCRIPTION
**Pull request by tilmantroester**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -111,7 +114,7 @@ endif()
 # The reason for building ccl as a shared library is that we can link it to
 # class directly, and it's not a dependency anymore
 add_library(ccl SHARED $<TARGET_OBJECTS:objlib>)
-target_link_libraries(ccl ${GSL_LIBRARIES} ${FFTW_LIBRARIES} ${CLASS_LIBRARIES} ${ANGPOW_LIBRARIES} m)
+target_link_libraries(ccl ${GSL_LIBRARIES} ${FFTW_LIBRARIES} ${CLASS_LIBRARIES} ${ANGPOW_LIBRARIES} m $<$<BOOL:${OpenMP_C_FOUND}>:OpenMP::OpenMP_C>)
```
Comment (damonge) [inline CMakeLists.txt:117]: OK, this notation is very obscure to me. What's going on?

**Review by damonge — COMMENTED** [review]
I think it looks ok, but can I check how this relates to #1018 and #1016? 
Should the branch associated with #1018 be deleted?

File: CMakeLists.txt (hunk 1 above)
Comment (tilmantroester) [reply CMakeLists.txt:118]: This is a generator expression (https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html). I wish there were an easier way to do this but with cmake, nothing ever seems easy to do. I added a comment on what it does.

This was in earlier versions already (e.g. 2.5.1) and just got restored here.

**Review by damonge — APPROVED** [review]
LGTM!

**Pull request merged**